src/App.tsx

```diff
@@ -1,223 +1,220 @@
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import {
+  addTodos,
+  deleteTodo,
+  getTodos,
+  patchTodo,
+} from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { ErrorNotification } from './components/Page/ErrorNotification';
+import { FilterTodos } from './components/Page/FilterTodo';
+import { Header } from './components/Page/Header';
+import { TodoList } from './components/Page/TodoList';
+import { TodoContext } from './components/TodoContext';
+import { TodosError } from './types/ErrorEnum';
+import { FilterType } from './types/FilterTypeEnum';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useContext(TodoContext);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.All);
+  const [title, setTitle] = useState('');
+  const [todosError, setTodosError] = useState<TodosError>(TodosError.None);
+  const [isAdding, setIsAdding] = useState(false);
+
+  if (todosError.length > 0) {
+    setTimeout(() => {
+      setTodosError(TodosError.None);
+    }, 3000);
+  }
+
+  const loadTodos = useCallback((userId: number) => {
+    getTodos(userId)
+      .then(todosFromServer => {
+        setTodos(todosFromServer);
+      })
+      .catch(() => setTodosError(TodosError.Loading));
+  }, [user, todos]);
+
+  useEffect(() => {
+    if (!user) {
+      return;
+    }
+
+    loadTodos(user.id);
+  }, [user]);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
-  }, []);
+  }, [todos]);
+
+  const visibleTodos = filterType === FilterType.All
+    ? todos
+    : todos.filter(({ completed }) => {
+      switch (filterType) {
+        case FilterType.Active:
+          return !completed;
+        case FilterType.Completed:
+          return completed;
+
+        default:
+          throw new Error();
+      }
+    });
+
+  const handleChooseFilter = useCallback(
+    (filter: FilterType) => {
+      setFilterType(filter);
+    },
+    [filterType],
+  );
+
+  const handleSubmit = useCallback(async (event: FormEvent) => {
+    event.preventDefault();
+
+    setIsAdding(true);
+
+    if (!title.trim()) {
+      setTodosError(TodosError.Title);
+      setTitle('');
+
+      return;
+    }
+
+    try {
+      if (!user) {
+        return;
+      }
+
+      const newTodo = await addTodos(user.id, title);
+
+      setTodos([...todos, newTodo]);
+    } catch {
+      setTodosError(TodosError.Adding);
+    }
+
+    setTitle('');
+    setIsAdding(false);
+  }, [title, user]);
+
+  const handleDelete = useCallback(async (todoId: number) => {
+    try {
+      await deleteTodo(todoId);
+
+      setTodos([...todos.filter(({ id }) => id !== todoId)]);
```

Comment:
Still need to fix:
If the new state is computed using the previous state, you can pass a function to setTodos